docs/usage/models.md

```diff
@@ -37,7 +37,8 @@ user = User(id='123')
 `user` here is an instance of `User`. Initialisation of the object will perform all parsing and validation,
 if no `ValidationError` is raised, you know the resulting model instance is valid.
 ```py
-assert user.id == 123
+# Note that 123.45 was casted to an int and its value is 123
```

Comment:
I didn't get the note here. The value that is provided for `id` field is `'123'` (`user = User(id='123')`). I don't know where does the `123.45` come from?

Also, there is a [Data Conversion section](https://pydantic-docs.helpmanual.io/usage/models/#data-conversion) in the doc that explains what is happening in casting the input data.

Reply:
`123.45` comes from the next line; see line [#41](https://github.com/samuelcolvin/pydantic/pull/4231/files/f225e480d5d2e7a061933153388bc0a6b8b4de68#diff-2e8eff0d0ff490b55e7f1bab8454001906bd22a5a3a99e89dc100132c57411fdR41); I changed the statement so the assertion indicates the change of type.

Reply:
Furthermore, as a new user of `pydantic`, I understand that the parsing vs. validating functionality is a source of confusion. Therefore, I found this example to be a good opportunity to stress the behavior.